README.md

```diff
@@ -93,9 +93,129 @@ TODO aspect ratio + images alignment + more more images
+
+Built-in support for capturing frames and debugging with Renderdoc.
+
+```c++
+const IQueue::SSubmitInfo submitInfo = {
+    .waitSemaphores = {},
+    .commandBuffers = {&cmdbufInfo,1},
+    .signalSemaphores = {&signalInfo,1}
+};
+m_api->startCapture(); // Start Renderdoc Capture
+queue->submit({&submitInfo,1});
+m_api->endCapture(); // End Renderdoc Capture
+```
+
+### 🧩 Nabla Event Handler: Seamless GPU-CPU Synchronization
+
+Nabla Event Handler's extensive usage of [Timeline Semaphores]() enables CPU Callbacks on GPU conditions.
+
+You can enqueue callbacks that trigger upon specific GPU conditions, enabling tasks like resource deallocation to be handled only after the GPU has completed the relevant work.
+```c++
+// This doesn't actually free the memory from the pool, the memory is queued up to be freed only after the `scratchSemaphore` reaches a value a future submit will signal
+memory_pool->deallocate(&offset,&size,nextSubmit.getFutureScratchSemaphore());
+```
+
+### 🧩 GPU Object Lifecycle Tracking
+
+Nabla uses [smart reference counting]() to track the lifecycle of GPU objects. Descriptor sets and command buffers are responsible for maintaining reference counts on the resources (e.g., buffers, textures) they use. The queue itself also tracks command buffers, ensuring that objects remain alive as long as they are pending execution. This system guarantees the correct order of deletion and makes it difficult for GPU objects to go out of scope and be destroyed before the GPU has finished using them.
+
+### 🧩 HLSL2021 Standard Template Library
+
+- 🔄 Reusable: Unified single-source C++/HLSL libraries eliminate code duplication with reimplementation of STL's `type_traits`, `limits`, `functional`, `tgmath`, etc.
+
+- 🐞 Shader Logic, CPU-Tested: A subset of HLSL compiles as both C++ and SPIR-V, enabling CPU-side debugging of GPU logic, ensuring correctness in complex tasks like FFT, Prefix Sum, etc.
+Future Proof: C++20 Concepts in HLSL for safe and documented Static Polymorphism
+
+- 🔮 Future-Proof: C++20 concepts in HLSL enable safe and documented polymorphism.
+
+- 🧠 Insane: Boost Preprocessor and Template Metaprogramming in HLSL!
+
+- 🛠️ Real-World Problem Solvers: The library offers GPU-optimized solutions for tasks like Prefix Sum, Binary Search, FFT, Global Sort, and even emulated `shaderFloat64` when native GPU support is unavailable!
+
+```
+[TODO][CODE] Code for each or just one showcasing most of the above points?
+```
+
+### 🧩 Full Embrace of [Buffer Device Address]() and [Descriptor Indexing]()
+
+By utilizing Buffer Device Addresses (BDAs), Nabla enables more direct access to memory through 64-bit GPU virtual addresses. Synergized with Descriptor Indexing, this approach enhances flexibility by enabling more dynamic, scalable resource binding without relying on traditional descriptor sets.
+
+### 🧩 Minimally Invasive Design
+[TODO]: vulkan handle acquisition, multiple windows, content playing second fiddle
```

Comment:
maybe show screenies of Nabla window and Nabla diff window on n4ce?

Write that its great for porting legacy apps from OpenGL and DirectX to Vulkan.

Reply:
the thing to highlight is probably our headless compute and graphics, as opposed to lets say SDL or GLFW where its assumed you'll have one window and a main thread.